### PR TITLE
Support payForTx, typed data, delegation signing with a basic confirmation

### PR DESCRIPTION
**Pull request by davidyuk**
This PR is supported by the Æternity Foundation

I've tested it using https://docs.aeternity.com/aepp-sdk-js/v13.2.2/examples/browser/aepp/

Take over this PR to implement confirmations as you need 🙏 

closes https://github.com/superhero-com/superhero-wallet/issues/2222

### CODE REVIEW

**Comment by Liubov-crypto** [comment]
LGTM

File: src/protocols/aeternity/libs/AeAccountHdWallet.ts [hunk 1]
```diff
@@ -82,16 +77,123 @@ export class AeAccountHdWallet extends AccountBase {
...
+    aci: Parameters<AccountBase['signTypedData']>[1],
+    options: Parameters<AccountBase['signTypedData']>[2] = {},
+  ): Promise<Encoded.Signature> {
+    if (options?.aeppOrigin) {
+      const dataType = new TypeResolver().resolveType(aci);
+      const decodedData = new ContractByteArrayEncoder().decodeWithType(data, dataType);
+      const {
+        name, version, networkId, contractAddress,
+      } = options;
+      const opt = {
```
Comment (peronczyk) [inline src/protocols/aeternity/libs/AeAccountHdWallet.ts:101]: I suggest to use more descriptive name instead of shortening the argument.

**Review by peronczyk — APPROVED** [review]

**Review by CedrikNikita — CHANGES_REQUESTED** [review]
This code is should be rewritten to not depend on sing message permissions. Because connected dapp in the current implementation with **silently** ask for delegations, if the auto sign message is turned on in the permissions.

**Review by CedrikNikita — APPROVED** [review]

File: src/protocols/aeternity/libs/AeAccountHdWallet.ts [hunk 2]
```diff
@@ -89,6 +94,96 @@ export class AeAccountHdWallet extends MemoryAccount {
...
+    if (options?.aeppOrigin) {
+      const params = unpackDelegation(delegation);
+      switch (params.tag) {
+        case DelegationTag.AensName:
+          try {
+            resolvedName = (await (await getMiddleware()).getName(params.nameId)).name;
+          } catch (e) {
+            handleUnknownError(e);
+          }
+          message = `sign delegation of ${resolvedName || params.nameId} to ${params.contractAddress}`;
```
Comment (davidyuk) [inline src/protocols/aeternity/libs/AeAccountHdWallet.ts:152]:
```suggestion
          message = `sign delegation of ${resolvedName ?? params.nameId} to ${params.contractAddress}`;
```

File: src/protocols/aeternity/libs/AeAccountHdWallet.ts [hunk 3]
```diff
@@ -89,6 +94,96 @@ export class AeAccountHdWallet extends MemoryAccount {
...
+          break;
+        case DelegationTag.AensPreclaim:
+          message = `sign delegation of name preclaim to ${params.contractAddress}`;
+          break;
+        case DelegationTag.AensWildcard:
+          message = `sign delegation of name claiming to ${params.contractAddress}`;
```
Comment (davidyuk) [inline src/protocols/aeternity/libs/AeAccountHdWallet.ts:158]:
```suggestion
          message = `sign delegation of all names management to ${params.contractAddress}`;
```
> AENS_CLAIM, AENS_UPDATE, AENS_TRANSFER, AENS_REVOKE

https://github.com/aeternity/protocol/blob/8a9d1d1/contracts/fate.md#appendix-2-delegation-signatures

File: src/protocols/aeternity/libs/AeAccountHdWallet.ts [hunk 4]
```diff
@@ -89,6 +94,96 @@ export class AeAccountHdWallet extends MemoryAccount {
...
+      );
+      if (!permissionGranted) {
+        throw new RpcRejectedByUserError();
+      }
+    }
+
+    return super.signDelegation(delegation, {
+      ...options, // Mainly to pass the `fromAccount` property
+      networkId: this.nodeNetworkId.value,
+      aeppOrigin: undefined,
```
Comment (davidyuk) [inline src/protocols/aeternity/libs/AeAccountHdWallet.ts:183]:
```suggestion
      networkId: this.nodeNetworkId.value,
```
MemoryAccount won't pass these options to `sign` method
https://github.com/aeternity/aepp-sdk-js/blob/2743e4af828902d90e532447b6c526c833869e49/src/account/Memory.ts#L207-L217

**Comment by Liubov-crypto** [comment]
I can delegate signature. 
But not all functions will work, for example if I want Generate spend transaction or Sign inner transaction or Pay for transaction , Deploy or Call Contract here https://docs.aeternity.com/aepp-sdk-js/v13.2.2/examples/browser/aepp/  
I will get an error: UnsupportedVersionError: Unsupported node version 7.0.0. Supported: >= 6.2.0 < 7.0.0

**Comment by Liubov-crypto** [comment]
I'm still getting an issue with UnsupportedVersionError: Unsupported node version 7.0.0. Supported: >= 6.2.0 < 7.0.0

geni
sp

**Comment by Liubov-crypto** [comment]
LGTM, tested in testnet

**Pull request merged**